virttest/utils_cpi.py

```
@@ -0,0 +1,934 @@
+        kernel_parts = guest_distro_info["kernel_parts"]
+        if (
+            parsed_level["kernel_major"] != kernel_parts["major"]
+            or parsed_level["kernel_minor"] != kernel_parts["minor"]
+            or parsed_level["kernel_stable"] != kernel_parts["stable"]
+            or parsed_level["patch_level"] != kernel_parts["patch_level"]
+        ):
+            raise AssertionError(
+                f"Kernel version mismatch: expected "
+                f"{kernel_parts['major']}.{kernel_parts['minor']}.{kernel_parts['stable']}-{kernel_parts['patch_level']}, "
+                f"got {parsed_level['kernel_major']}.{parsed_level['kernel_minor']}.{parsed_level['kernel_stable']}-{parsed_level['patch_level']}"
+            )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)